### ISSUE TITLE
[D8][UX] Rename the "Locale" module to "Interface Translation"

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#4562]
...just the human-readable name of course - machine name stays the same, so drush commands and code that mentions `locale` should still work the same.

- "Locale" as a word means absolutely nothing to people, unless they are speaking a Germanic or French language (and even then, not sure), or if they are technology buffs.
- Even the English definition of the word does not make sense in the context that we are using it (Oxford):
  > a place where something happens or is set, or that has particular events associated with it.
- This is basically the same principle as with the decision to minimize the use of words such as "entities" and "bundles" in the admin UI, as well as to not mention "node" and use "piece of content" or "content item" instead: because people do not understand "node"/"entity"/"bundle" unless they are developers or long-time Drupal/Backdrop users.
- We already have a "Content Translation" module that provides the functionality to allow translation of content. This one is its counterpart for translating strings in the admin interface. And since it is about interface translation, we should call it "Interface Translation".

They've done the same in D8:

![image](https://user-images.githubusercontent.com/2423362/90935392-c5a63a00-e40b-11ea-9c04-4a54d76476b4.png)

While at it, the description of the module currently is this:

> Provides language negotiation functionality and user interface translation to languages other than English.

I am proposing this instead:

> Provides language negotiation functionality, as well as translation of the administrative interface text to other languages.

I would like to refine "language negotiation" as well, because it sounds too technical, but I don't have anything better to propose at the moment.

Finally, we should make sure to retain the current functionality, in which when people explicitly search for "locale" in the module listing, then the module still comes up in the results. Switching the labels of the modules in the module listing to be using `theme_label_machine_name()` achieves that, and it also standardizes things and makes them consistent with other listings in the admin UI:
<img width="1191" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/c8cd84d3-a96d-4849-94fc-bb53c14284d4">

**Comment by jenlampton** [comment]
Love it!

**Comment by klonos** [comment]
...and there's a PR for it now 😉

https://github.com/backdrop/backdrop/pull/3270

**Comment by klonos** [comment]
...I just tested this in the PR sandbox, and the module filter doesn't work if you type "locale". I'll see if I can fix that then 🤔

**Comment by indigoxela** [comment]
Sorry for being rather late here...

I do have three small concerns:

1. In general it's easier for developers, if the module's human-readable name is similar to the machine name
2. "Interface Translation" does not cover our Configuration Translation - we also use locale module for that. (Things like site_name, site_slogan...)
3. People coming from D7 might search for "Locale" in the UI, as they're already familiar with that name - some of them might be baffled, if they don't find it

> the module filter doesn't work if you type "locale"

If that can get solved, and (at least) the description also notes _config_translatables in some way,  I'm fine with that change.

**Comment by klonos** [comment]
> In general it's easier for developers, if the module's human-readable name is similar to the machine name

I don't fully agree on that point. What matters to devs is the machine, and that won't change. The actual module name used in the admin UI is for site builders/admins and Backdrop evaluators though, so it should have a name that these people understand. If I wanted to translate something, I would start typing "transl..." in that filter, and then my eyes would scan for modules that have that term in their name. Locale seems odd, and since I'm not a native English speaker, I only came across that term in Drupal-land 🤷 

> "Interface Translation" does not cover our Configuration Translation...

That's a very valid point. Suggestions on how to deal with that? ...perhaps add something in the description text?

>People coming from D7 might search for "Locale" in the UI...

That's true. We need to make that filter work with machine names too. I'll take a stab at it.

**Comment by jenlampton** [comment]
> perhaps add something in the description text?

What about adding something like "Formerly the Locale module..." at the beginning?

**Comment by klonos** [comment]
> People coming from D7 might search for "Locale" in the UI...

With #3402 allowing for a label/machine name combo via the `theme_label_machine_name()` function, should we use that to expose machine names in the modules listing as well? If we did that, we'd be able to have the new "Interface Translation" label + the old `locale` machine name.

**Comment by klonos** [comment]
...aaaaand, with #3402 done and dusted, things are now easy 🙂 

New PR here: https://github.com/backdrop/backdrop/pull/4779

<img width="1191" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/c8cd84d3-a96d-4849-94fc-bb53c14284d4">

- renames the human-readable name of the module to "Interface Translation"
- changes the label column to use `theme_label_machine_name()`, which exposes the machine name to the UI + allows showing the module in the results when searching for "locale" (as shown in the screenshot above)
- changes the module description from this:
  `Provides language negotiation functionality and user interface translation to languages other than English.`
  ...to this:
  `Provides language negotiation functionality, as well as translation of configuration and the administrative interface to other languages.`

@indigoxela I believe that that addresses all your previous concerns. Can you please check and let me know?

@jenlampton @stpaultim @olafgrabienski (and others too) you might wanna test and chime in.

The reason I changed the `languages other than English` bit is because the UI can actually be "translated" into English as well (see [this thread in Zulip](https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/.E2.9C.94.20Stringoverrides.20or.20equivalent/near/443385692)), and this sentence does not make too much sense when the default language of the site is not English. So "other than" wasn't 100% accurate previously.

Doing that, also intentionally removed the word "English", which has often been mentioned to be single-sided. https://www.drupal.org/project/drupal/issues/3261053 is one of the most recent examples that I can recall, but there have been other cases where people have been asking for the Drupal UI to be more language-agnostic and less English-specific where possible. I believe that the description change addresses that.

**Comment by klonos** [comment]
...the PR is currently suffering from a malicious comment added by a user account that must have found some exploit in GitHub 😞 ...I've reported the specific user account and blocked them, but the comment is still there and is disruptive ...it causes the entire page to blink with a message, and blocks access to the main PR thread (so no access to the PR sandbox). Till that is sorted, please use the following links:
- to review the code changes: https://github.com/backdrop/backdrop/pull/4779/files
- to test the PR sandbox: https://pr4779-gusw0w2vg4yoac1rvluq8fg8lkhxlsqe.tugboatqa.com (password is `59eb7cb6a490`)

**Comment by indigoxela** [comment]
@klonos which problem are we trying to solve by obscuring the module display name to something that doesn't match anything in the modules directory and is different to what we use with bee or with any hook implementation? :wink:

Because "Drupal did..." isn't a valid argument in any way.

Adding the milestone was IMO premature. I don't think we can assume consensus here.

**Comment by findlabnet** [comment]
It seems completely unnecessary to confuse the term "Locale" commonly used in computing.

**Comment by klonos** [comment]
> ... that doesn't match anything in the modules directory and is different to what we use with bee or with any hook implementation

> It seems completely unnecessary to confuse the term "Locale" commonly used in computing.

I really question these arguments and object them VIVIDLY! Here's a quick TL;DR:
- We are already doing that with many other modules in core (see details below)
- This is about the UI/UX and user-facing admin text - it is NOT about any command-line tool, developers, nor the code behind it. Can I please ask both of you to revisit https://backdropcms.org/philosophy and pay attention to the bold text at the top of the page:
  > Backdrop values the needs of the editor and architect over the needs of the developer.
- This issue here is not to suggest a change to the machine name of the module. Nothing will break for developers or people using bee.

There are actually several reasons/points for this change actually:
- End users will not care what the files in the codebase are called, nor what names the respective function names have. We are not "obscuring" anything. Developers can still find the module name very easily. We are NOT changing that - we are actually indicating it in the user interface to assist any of them that might be confused (which I doubt).
- "Locale" as a word means absolutely nothing to people, unless they are speaking a Germanic or French language (and even then, not sure).
- Even the English definition of the word does not make sense in the context that we are using it (Oxford):
  > a place where something happens or is set, or that has particular events associated with it.
- Module machine names and human-readable names do not match, nor do they need to match 100%. We have several rules around naming modules, however, none of the rules mandates matching the machine name 100% to the human-readable name. See https://docs.backdropcms.org/creating-modules. If you want to propose that as a requirement, then please feel free to raise a separate issue.
- We already have the following (which I'm sure you have noticed already, so I don't really understand why you are presenting the argument about the difference between machine name and human-readable):
  - `config` vs. "Configuration Manager"
  - `dblog` vs. "Database Logging"
  - `installer` vs. "Project Installer"
  - `syslog` vs. "System Logging"
  - `simpletest` vs. "Testing"
  - `update` vs. "Update Manager"
  - `translation` vs. "Content Translation"
  - `contextual` vs. "Contextual Links"

  So I don't understand why the big fuss about this specific case, when we have not one or two, but several examples already in core, without anyone ever complaining about any "obscuring" happening because of that.
- The bottom-line reason why we have these differences is always the same: developers often make poor name choices and they don't realize it until after people start using the product. We then adapt the UI based on user feedback, best UX practices, and also on what makes sense and makes the UI as intuitive as possible, without people having to resort to reading lengthy help text, documentation, or look up words in the dictionary.
- This is basically the same principle as with the decision to minimize the use of words such as "entities" and "bundles" in the admin UI, as well as to not mention "node" and use "piece of content" or "content item" instead: because people do not understand "node"/"entity"/"bundle" unless they are developers or long-time Drupal/Backdrop users.
- We already have a "Content Translation" module that provides the functionality to allow translation of content. This one is its counterpart for translating strings in the admin interface. And since it is about interface translation, we should call it "Interface Translation".

Let me know if the above points provide enough reasoning. If not, I can elaborate further.

Fair enough about the milestone - I removed it. I will bring this up during the next UX meeting before adding it back after consensus. To be honest though, I thought that we did have consensus:
- @jenlampton said that she loved it.
- you also said:
  > If that can get solved, and (at least) the description also notes _config_translatables in some way, I'm fine with that change.

Actually, now that I think about it with regards to the suggestion to be mentioning "_config_translatables": the current description does not include that, nor does the current name of the module imply it in any way. You have to be a person that knows the internals of the code to know all that. Since we are not mentioning `t()` or any other ways that translatable strings are provided, then I don't think that we should be mentioning "_config_translatables" either. End users simply won't care. So I find the following a much simpler description that we should use instead (I've updated the PR to be using that):
> Allows translation of the administrative interface text to other languages.

> Because "Drupal did..." isn't a valid argument in any way.

I will follow up with a private message directly to you in Zulip with regards to this, because I feel that it is getting out of hand, and I don't want to derail this issue.

**Comment by klonos** [comment]
> Because "Drupal did..." isn't a valid argument in any way.

I will follow up with a private message directly to you in Zulip with regards to this, because I feel that it is getting out of hand, and I don't want to derail this issue.

**Comment by findlabnet** [comment]
For me, the second result in a Google search for "locale" is [Locale](https://en.wikipedia.org/wiki/Locale_(computer_software))

> Backdrop values the needs of the editor and architect over the needs of the developer.

Do not assume that people are completely illiterate (if true, they do not use CMS and are not the target audience).

In my opinion discussions should be less emotional, sorry if it is because of me.

**Comment by kiamlaluno** [comment]
I would say that all the module names listed in the previous comment do not make people think the module purpose is different from its effective purpose.  
The Locale module is not only about interface translation; it should not be renamed *Interface translation*, IMO.

**Comment by laryn** [comment]
I think this clarifies things for new users, and while I appreciate the concerns, I don’t find that it obscures the machine name — in fact explicitly showing the machine name in the standard way is a great touch. I like it.

I might not always think of the same things as @klonos  but I absolutely appreciate his ceaseless drive to simplify things for new users. So, thank you, @klonos!

**Comment by klonos** [comment]
> The Locale module is not only about interface translation; it should not be renamed Interface translation, IMO.

That's fair to say @kiamlaluno. What do you propose then?

BTW, the description of the module currently is this:
> Provides language negotiation functionality and user interface translation to languages other than English.

I am proposing this instead:
> Provides language negotiation functionality, as well as translation of the administrative interface text to other languages.

I would like to refine "language negotiation" as well, because it sounds too technical, but I don't have anything better to propose at the moment.

**Comment by klonos** [comment]
> Do not assume that people are completely illiterate (if true, they do not use CMS and are not the target audience).

@findlabnet @indigoxela and I had a very constructive discussion in private. This is what I explained to her:
> Well, showed this to my 12yo daughter while teaching her Drupal/Backdrop:
> <img width="445" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/7d25857f-d6f8-4b6f-a646-3e883630fec3">
 
> She is not a computer buff, but is tech-savvy enough for her age. She also speaks English fluently, but that is a second language to her. Given her young age, and the fact that she only started learning English at 6yo, her vocabulary is not that developed yet. So if you tell her "interpolation" or "assimilation" or any fancy words like that, she might not know all of them.
> 
> Given the above, she understood straight away what "Content Translation" is, as well as what "Language" is, without even having to read their descriptions. Then she said that she doesn't know what "Locale" is ...and I then told her the same thing I always tell her: "google it" ...and then she saw this:

> <img width="847" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/0b0c9154-0559-45de-929e-6774ee56dff5">

> ...which made things even worse. Now I had to go explain so many things to her.
> 
> I still haven't shown her the UI in the PR to get her feedback if she finds it more straight-forward, but I'm sure based on our discussion that it would be,

**Comment by klonos** [comment]
> In my opinion discussions should be less emotional, sorry if it is because of me.

This was partially me reading things the wrong way. @indigoxela pointed out the fact that the issue description here is very "spartan", so when it also says "They've done the same in D8", it makes it sound like that's the only argument for suggesting this change. I will move some of the bullet points in my earlier comment in the issue summary to try and make it clear.

**Comment by klonos** [comment]
...I've updated the issue summary, so I hope that the proposal is less confusing now.

**Comment by findlabnet** [comment]
@klonos I very much appreciate your daughter's opinion (seriously), but I'm still confused by some things.

> because it sounds too technical

Who is it right for (maybe 80% of respondents)?
Where is the limit for "too..."? 
Who has the right to decide it is "too technical", "less technical" or even just stupid?
And why is it so bad to call things by their name?

By the way, for me English is only a third of the languages I use every day and my level in it is very basic.

**Comment by klonos** [comment]
My daughter did not express any opinion on this, nor did she suggest any change - she was just confused when I started showing here how to enable multilingual features in Backdrop. The process was like this:
- These features do not come enabled by default, so lets enable some required modules.
- Go to the modules page, then filter by "language" in order to show all language-related modules that we need.
- Let me explain what these 3 modules do and why we need each one for...

And that's the point when she started being confused with "locale". She said "I know _local_, but I've never seen locale _before_".

> Who has the right to decide it is "too technical", "less technical" or even just stupid?

I am finding it hard to answer this the way that it has been phrased. There's plenty of people in the community that can make a fair judgement. Some of us have been following up on UX and UI design (outside of Drupal/Backdrop), and some of us have even worked for many years as trainers (either in IT, or specifically training people in Drupal). Some notable examples:
- https://www.drupal.org/u/jenlampton
  > Jen Lampton spent several years as the Director of Drupal training at Chapter Three in San Francisco. In that role she developed Drupal workshop materials, and used them to train hundreds of people all across the US on how to use and extend Drupal.
- https://web.archive.org/web/20230303005121/https://salsa.digital/about/team/greg-netsas
  > Greg also conducts many of Salsa’s GovCMS and Drupal training sessions, running our content management and site builder courses.

The above are not the only ones with relevant experience, and certain not "authorities" on the matter. I would say that everyone can express their thoughts and make suggestions to improve things.

I hope that the above provides some clarification.

**Comment by findlabnet** [comment]
@klonos I do not doubt your or Jen's expertise at the highest level. 
But my simple questions (as it was phrased) still need simple answers. 
Unfortunately, we have too many examples of bad things happening when only someone knows the right direction.

**Comment by klonos** [comment]
@findlabnet someone needs to make a decision to make things happen. We aren't always making the right decisions and we cannot all agree on everything all the time, but we can always change/improve further if it turns out that something was wrong or inadequate.

Unfortunately, this discussion is derailing progress on this issue. Can we please keep comments here specific to the task, and discuss generic decision-making process etc. elsewhere?

**Comment by findlabnet** [comment]
> someone needs to make a decision

Who is it and why are you? 
At least two or three people have said NO, but you insist on doing it, whatever it takes, according to your own judgment.

It wouldn't occur to my 12 year old granddaughter to install and configure a CMS, but I know I can show and explain it to her without discounting her non-technicality (she's very far from technical) and she'll understand it at a glance, even though English is her second language (she's fluent, having started learning it at age 6). 
She just realizes that you can't break things just because things are called that, not the other way around.

**Comment by klonos** [comment]
@findlabnet did you see this?:
> Can we please keep comments here specific to the task, and discuss generic decision-making process etc. elsewhere?

**Comment by klonos** [comment]
...I have also started a thread specifically about discussing project name policy here: https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/Policy.20about.20project.20names/near/443526007

**Comment by stpaultim** [comment]
> Do not assume that people are completely illiterate

I consider myself to be pretty literate, yet I've never really understood what the "Locale" module is for. I've had it explained to me, after which it makes sense for a while, but by the next time I run across it, I've usually forgotten. Given the context of the discussion, I can figure out what this means "Provides language negotiation functionality." But, the words alone mean nothing to me and I'm considering writing a contrib module to provide a translation for this jargon, which I find meaningless :-). 

(Sorry, for the jokes).

I agree that "Interface Translation" is not the best label, but in my opinion it's much better (by "better", I mean "meaningful") than "Locale", in my opinion. I'll brainstorm some options, none of which are perfect.

"Internal Translation" - does this cover both interface and configuration
"System Localization" - it's not as descriptive as other options, but it is closer to the machine name and better than the current name. 
"System Translation" - does this include all the internal things it does?
"Interface & Config Translation" - I know, it's probably too long. But, I could live with that if others could.

I understand that changing item labels to something that does not match the machine name is controversial. One of my most involved core PR's changed the label for "Full HTML" to "Raw HTML" while keeping the old machine name. We had the same arguments at that time.
https://github.com/backdrop/backdrop-issues/issues/4499

We did something similar when we changed "Filtered HTML" to "Basic." 
https://github.com/backdrop/backdrop-issues/issues/1188

In those cases, and others, we believed that the improvements to the UI were great enough to justify the inconvenience of mis-matched machine names. 

I think that listing machine names on the modules page does help in this regard.

In summary, I would not block this PR because it makes things a bit more difficult for developers, if it's a substancial improvement for people using Backdrop that are not developers. I think that it probably is worth it and our willingness to make these kinds of changes is one of the things that motivates me to keep contributing. 

This change is not a priority for me, but I do think it's an improvement. I look forward to hearing from some more of our core contributors (thanks to @laryn for chiming in). If we can improve the language a bit, I'd be in favor of it.

**Comment by klonos** [comment]
> does this cover both interface and configuration ... "Interface & Config Translation"

Both of these are affecting strings that will be presented in the user interface. The distinction between "configuration translation" and the rest of the "regular" translation only makes sense to developers. End users will still see the text in the admin UI, regardless of whether it is provided by `t()` or by config strings in .json files. So the question is: does it really matter differentiating these when they are both achieving the same end result?

Also `Internal Translation` doesn't seem right. "Internal" as opposed to what? What would be "External Translation" then?

I think I like "System Localization" or "System Translation" better than "Internal Translation", but not 100% sure. I have to think a bit further if they convey the correct meaning, or if they leave room for possible confusion/ambiguation. Curious to see what others think about these two proposals.

**Comment by kiamlaluno** [comment]
I would avoid *System* in a module name that is shown in admin/modules/list: People just skip that word and read the module name as *Localization* and *Translation* instead of *System Localization* and *System Translation*.  
*Trenslation* does not suite the module since it does not just translate. If we go with the same reason not to use °Locale*, *Localization* should be avoided too, since the *Oxford Advanced American Dictionary*, for *localization* shows:

> **localize something** to limit something or its effects to a particular area
> **localize something** (*formal*) to find out where something is

**Comment by kiamlaluno** [comment]
As for the suggested module description (*Provides language negotiation functionality, and allows translation of the administrative interface text to other languages.*) the last part it is better left as the current description: *allows translation […] to other languages* translated in Italian would become *permette la traduzione ad altri linguaggi* which (since read in Italian) would mean *allows translation […] to languages different from Italian* (for example, from Italian to English, from Italian to French), which is not how the translation works.

**Comment by stpaultim** [comment]
> Also Internal Translation doesn't seem right. "Internal" as opposed to what? What would be "External Translation" then?

I was thinking that external translation is "content translation". Content is what people come to see at the site and might "figuratively" be considered exteral to the workings of the site.
Internal translation would be translation of what is inside the site, namely configuration and interface. 

Just grasping for something that might be meaningful, if not precise. 

Again, it's not perfect. I'd love something better. But...

> Translation does not suite the module since it does not just translate.

@kiamlaluno - Then what does the module do? I'm sincerely confused and baffled. I certainly went into this discussion confused about the purpose of this module. I thought I understood that it's a tool we use to translate the interface and configuration. Now, you suggesting it does more. What else does it do?

**Comment by kiamlaluno** [comment]
@stpaultim The description for the `@file` tag says that: ***Add language handling functionality** and enables the translation of the user interface to languages other than English.*  
Also its .info file does not say the module just translate the user interface: *Provides **language negotiation functionality** and user interface translation to languages other than English.*

**Comment by kiamlaluno** [comment]
Truly, having a Language module and a Locale module is not the best option.  
It would have been better to leave the language part to the Language module and have a module to translate the interface, instead of the Locale module. In that case, *Interface Translation* would have been the perfect name for the module.

**Comment by klonos** [comment]
> ...which is not how the translation works.

Thanks @kiamlaluno 🙏🏼 ...that is really helpful. I realize now something that is obvious but I somehow failed to consider: that the "default" source strings in the code and configuration are in fact _always_ English. So the current "other than English" part of the description makes 100% sense 👍🏼 

> I was thinking that external translation is "content translation". Content is what people come to see at the site and might "figuratively" be considered exteral to the workings of the site.

OK. I see the reasoning behind that suggestion now 👍🏼 ...it wasn't apparent to me though - only after it was explained.

>> Translation does not suite the module since it does not just translate.
>
> @kiamlaluno - Then what does the module do? ...

I'm glad some actually asked @stpaultim ...here's what I have known historically from d.org:
- https://www.drupal.org/project/drupal/issues/1293304 (with regards to what the Locale module does, and whether "locale" is an appropriate term to use)
  > - Locale module does too much: manages list of languages, the selection from them in bootstrap/context, a UI translation feature and mass-import/export for that. We want to expand this in D8 with localization update module functionality (think automated feeding of the mass-import with up-to-date files from localize.drupal.org).
  > - Locale is a bad name. Locale module in fact does not handle any of the things that the industry calls locale, like timezones, currency formatting, first day of week, etc. Drupal handles these at other places and those are not language/culture dependent.
  > - We might or might not add (more) actual locale functionality in D8 core, so we might or might not going to have functionality in core that we can rightly call locale in terms of general industry naming conventions.
- https://www.drupal.org/project/drupal/issues/1222106 (with regards to the term "language negotiation")
  > Drupal uses language negotiation, selection, detection and providers to name things that are often overlapping.

  In Drupal, they've made big API-breaking changes in order to fix that "properly". I don't see how we could do that in Backdrop. That's why I thought that perhaps we can see of we can come up with a plain-English term that makes more sense.

So @kiamlaluno although I agree with what you are saying with regards to "it doesn't do just translation", we cannot possibly describe everything that this module does in a practical way - not in its name/label at least. That's what we have the description for. Right?

Initially, I thought that perhaps "Language handling" might be a better plain-English phrase than "Language negotiation", but then when I see that from the PoV of a person new to Drupal/Backdrop, I can't help but wonder "If that module handles language, then what is the Language module for?"

**Comment by klonos** [comment]
> ...I realize now something that is obvious but I somehow failed to consider: that the "default" source strings in the code and configuration are in fact _always_ English. So the current "other than English" part of the description makes 100% sense 👍🏼 

Actually hold on! ...the above might be true for strings coming from `t()`, but when it comes to config translation, the strings (such as the site name/slogan for example) might be in a language other than English.

Naming things is hard.

**Comment by klonos** [comment]
> Truly, having a Language module and a Locale module is not the best option.  
It would have been better to leave the language part to the Language module and have a module to translate the interface, instead of the Locale module. In that case, *Interface Translation* would have been the perfect name for the module.

I couldn't have described it better @kiamlaluno 👍🏼 ...that sounds like a 2.0 thing though (if we decided to do it), and definitely a separate issue.

**Comment by kiamlaluno** [comment]
> I can't help but wonder "If that module handles language, then what is the Language module for?"

The *Language* module is the one that handles the form on admin/config/regional/language, which essentially adds the languages to a site.

That is the real issue, not the module name: There should be the *Language* module, which includes the code that is now part of the *Locale* module, except for the interface translation code; then, there should be the *Interface Translation* module, which does exactly that.

If the plan is to first change the module name, and then move code from the *Locale* module to the *Language* module (as I described earlier), I am in with starting with changing the module name.

**Comment by klonos** [comment]
> If the plan is to first change the module name, and then move code from the Locale module to the Language module (as I described earlier), I am in with starting with changing the module name.

I think that I'm on-board with that too. I don't know if we have consensus for that though (and it still sounds like a 2.0 thing). I will mention it in the next dev meeting to get some feedback/thoughts.

**Comment by kiamlaluno** [comment]
Yes, moving code from a module to another is for the *2.x* branch. To me, that is the condition to accept the module name change, though.

**Comment by herbdool** [comment]
Naming is even harder when a module does multiple things. Another idea is to split it into two so there's then Language, Interface Translation, and Language Negotiation (or better wording).

Easier to name when a module does *one* main thing well. Or maybe that is too fine grained?

**Comment by argiepiano** [comment]
My 2 cents: renaming this module is a bad idea because:

- Locale will rarely be used "out of the box" by a site builder. The "learning curve" for multilingual sites is steep. It required me to read help articles and watch videos, and...
- ALL those existing articles and videos refer to a module called "Locale", which would no longer exist by that name in Backdrop

To me, what seems like a "clarifying step" by renaming this module will end up hurting everyone who needs to search for existing resources to learn how to set up multilingual sites.

That's the practical me, I guess.

**Comment by herbdool** [comment]
Good point @argiepiano. With that, I vote for just improving the description and documentation rather than renaming.

**Comment by kiamlaluno** [comment]
That could be said for any change done to code: Every introduced class, method, function, or hook would make the existing blogs obsolete. Even the existing modules could not work correctly, with the introduced changes.  
Yet, those changes are not considered material for the *2.x* branch. In fact, they are done in the *1.x* branch.

**Comment by argiepiano** [comment]
>  Every introduced class, method, function, or hook would make the existing blogs obsolete.... Yet, those changes are not considered material for the 2.x branch. In fact, they are done in the 1.x branch.

Yes, changes usually create friction. The key question is: is the change needed? Will it create more benefits than problems? Having worked with multilingual setups, I believe in this case, it doesn't.  The change is rather frivolous. In fact I'm surprised people are spending so much energy and emotional capital discussing this. This is my last post here :)

**Comment by docwilmot** [comment]
How about a simple compromise? Change the UI to say `"Locale (Interface translation)"`. Everybody wins

**Comment by kiamlaluno** [comment]
Also changing the names for the default text formats causes the same type of problems. They also involve more people, since people who must select a text format for a field are more than the people who will ever enable the Locale module.  
It will be more frivolous if the same type of changes is "randomly" applied.

**Comment by klonos** [comment]
> How about a simple compromise? Change the UI to say `"Locale (Interface translation)"`.

That's more or less what @jenlampton suggested [here](https://github.com/backdrop/backdrop-issues/issues/4562#issuecomment-683070353):

> What about adding something like "Formerly the Locale module..." at the beginning?

But that suggestion was before we had `theme_label_machine_name()`. With is, instead of adding the word "Locale" as a prefix to the module name or description, it is being added in the same consistent manner that other machine names are added elsewhere in the admin UI.

> ... which would no longer exist by that name in Backdrop

That is not what is happening here. The proposed PR does not _remove_ the name "locale", nor does it rename the module - it renames the human-readable label, however it still leaves it as a machine name. So "locale" is still visible and discoverable by searching with the term "locale" in the module list:
<img width="1191" alt="image" src="https://github.com/backdrop/backdrop-issues/assets/2423362/2e955c15-94cd-4672-bd1d-60a03551d2b2">

So the whole argument that people won't be able to find it does not apply.

In fact, if someone is trying to achieve interface translation and they search for "locale" after watching a video tutorial or reading up on documentation, they will get a result that has both things.

A couple of counter-arguments to that:
- Newer documentation is more likely to exist for newer versions of Drupal. If people happen to watch/read that, then they are more likely to find the term "Interface Translation". Then, when they search for that in Backdrop, they'd say "Meh, Backdrop doesn't have that" and give up. Having both "locale" (as a machine name) as wall as "Interface Translation" will cover all documentation.
- We also changed "Modules" to "Functionality", and "Themes" to "Appearance" (and all references of "users" to "accounts" or "people", and all references of "nodes" to "content etc."). All these years, I have never seen anyone complain that they couldn't find where modules and themes are in Backdrop, nor that they couldn't understand what "content items" are. Whereas with Drupal, we know what happened when people tried to learn what these things are. I cannot recall anything in the issue queue, nor the forum, nor the chat about people being confused by what these tings are, or ask why we changed their naming. In fact, it is more probable for people that are non-Drupal, new to the CMS concept and trialing Backdrop, to easily discover and understand what the "Functionality" menu provides, rather than understand what "Modules" are (which is a Drupalism, while the entire industry is using the terms "addon" or "plugin" or "extension").

So no, I am not buying the "this will make things harder" argument.

**Comment by klonos** [comment]
> ...I am not buying the "this will make things harder" argument.

As I am not buying the "developers will be confused if they don't find matching file names/structure" argument. Really?

**Comment by laryn** [comment]
One's perspective on this seems to depend on whether they are primarily looking back (existing developer's knowledge, old D7 training guides, etc.) or forward (newcomers to Backdrop CMS who have no knowledge of D7, hopefully making it easier to understand what modules do out of the box). I think the proposal strikes a good balance of being friendly to the future but also nodding to the past (with the machine name showing); I'm also not opposed to @docwilmot suggestion. 

That said, I'm also not interested in expending a lot of energy on this and other people feel much more passionately about this particular change than I do.
